flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/history/HistoryServerArchiveFetcher.java

```diff
@@ -153,9 +157,15 @@ public void run() {
 						continue;
 					}
 					boolean updateOverview = false;
+					long now = System.currentTimeMillis();
 					for (FileStatus jobArchive : jobArchives) {
 						Path jobArchivePath = jobArchive.getPath();
 						String jobID = jobArchivePath.getName();
+						if (retainedApplicationsMillis != 0L && now - jobArchive.getModificationTime() > retainedApplicationsMillis) {
```

Comment:
Could we assert that `retainedApplicationsMillis` is always no negative?

Reply:
Thank you for the review, change the default value to -1.

Reply:
Sorry for the ambiguous description, what I mean is "If a user sets a negative number, what will the behavior be?"
Maybe we should add a check logic

Reply:
Thank you for the review @klion26 , negative or zero value number is considered to disable the cleanup mechanism.
Add the hint statements in a log.warn fashion.